crates/egui/src/text_selection/text_cursor_state.rs

```diff
@@ -302,18 +302,42 @@ pub fn find_line_start(text: &str, current_index: CCursor) -> CCursor {
     // number of multi byte chars.
     // We need to know the char index to be able to correctly set the cursor
     // later.
-    let chars_count = text.chars().count();
+    let mut char_line_indexes: Vec<usize> = create_char_line_indexes(text);
+    let line_start_index = get_line_start_index(&mut char_line_indexes, current_index.index);
 
-    let position = text
-        .chars()
-        .rev()
-        .skip(chars_count - current_index.index)
```

Comment:
What was wrong with the old code?

Maybe it just needs to handle the `char_count < current_index.index` case?

Reply:
Since UTF-8 characters use 1 to 4 bytes,
using rev() can cause a panic and terminate the program.
You can test it by copying various UTF-8 characters
and using them in multiline().

Reply:
`.rev()` doesn't panic: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1647ea03b9ad85c69d72ee2189b73fc0

Reply:
Mix the UTF8 characters with English, fill the screen ( with them using multiline() ), and pressing Shift + TAB here and there.

Reply:
> Mix the UTF8 characters with English, fill the screen `( with them using multiline(), egui_demo_app )`, and pressing Shift + TAB here and there.

Reply:
It's sample TEXT.                                                                     
fill the screen.                                                                      
"    A äö 日本語 A äö 日本語"

Reply:
Mix the UTF8 characters with English, fill the screen ( with them using multiline() ), and pressing Shift + TAB here and there.

Reply:
It seems to happen when transitioning from using char index
to using an array of byte indexes.

There are two more instances of `rev()` in `text_cursor_state.rs`,
fortunately, those two do not cause panics
but instead incorrectly select a block where UTF8 characters exist.
I will register this as a separate issue later.

Reply:
@emilk
Have you noticed the panic occurring?

Reply:
I don't have time for this right now. I believe you that here is a panic in egui, but I don't think the panic is in the call to `.rev()`, and so I think a much simpler solution to your problem can be found.

Run in debug mode and RUST_BACKTRACE=1 and paste the stack-trace to the actual panic.